azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/FunctionCoreToolsHandlerImpl.java

```diff
@@ -34,22 +42,27 @@ public FunctionCoreToolsHandlerImpl(final AbstractFunctionMojo mojo, final Comma
     }
 
     @Override
-    public void installExtension() {
-        try {
-            final String localVersion = getLocalFunctionCoreToolsVersion();
+    public void installExtension(Set<Class> bindingTypes) throws Exception {
+        final String localVersion = getLocalFunctionCoreToolsVersion();
 
-            if (isLocalVersionSupportAutoInstall(localVersion)) {
-                installFunctionExtension();
+        if (isLocalVersionSupportAutoInstall(localVersion)) {
+            checkVersion(localVersion);
+            installFunctionExtension();
+        } else {
+            if (checkExtensionNecessity(bindingTypes)) {
+                this.mojo.error(NO_LOCAL_FUNCTION_CORE_TOOLS);
+                throw new Exception(NO_LOCAL_FUNCTION_CORE_TOOLS);
```

Comment:
What's the output if we going to this if-block?

Will the `NO_LOCAL_FUNCTION_CORE_TOOLS` be printed twice?